Sources/Tokenizers/Tokenizer.swift

```diff
@@ -256,6 +285,54 @@ public class PreTrainedTokenizer: Tokenizer {
     public func convertIdToToken(_ id: Int) -> String? {
         model.convertIdToToken(id)
     }
+    
+    public func applyChatTemplate(messages: [[String: String]]) throws -> [Int] {
+        try applyChatTemplate(messages: messages, chatTemplate: nil, addGenerationPrompt: true, maxLength: nil)
+    }
+    
+    public func applyChatTemplate(
+        messages: [[String: String]],
+        chatTemplate: String?,
+        addGenerationPrompt: Bool = false,
+        padding: Bool = false,
+        truncation: Bool = false,
+        maxLength: Int?
+    ) throws -> [Int] {
+        let template = try Template(chatTemplate ?? tokenizerConfig.chatTemplate?.stringValue ?? defaultChatTemplate)
+        var context: [String: Any] = [
+            "messages": messages,
+            "add_generation_prompt": addGenerationPrompt
+        ]
+
+        for (key, value) in tokenizerConfig.dictionary {
+            if specialTokenAttributes.contains(key), value != nil, !(value is NSNull) {
+                context[key] = value
+            }
+        }
+
+        let rendered = try template.render(context)
+
+        var encodedTokens = encode(text: rendered)
+
+        var maxLength = maxLength ?? encodedTokens.count
+
+        maxLength = min(maxLength, tokenizerConfig.modelMaxLength?.intValue ?? maxLength)
+
+        if encodedTokens.count > maxLength {
+            if truncation {
+                encodedTokens = Array(encodedTokens.prefix(maxLength))
+            }
+        } else {
+            if padding {
+                encodedTokens = encodedTokens + Array(
+                    repeating: model.padTokenId ?? model.eosTokenId ?? 0,
+                    count: encodedTokens.count - maxLength
+                )
+            }
```

Comment:
I wonder if we could ignore padding for now and simplify things a bit, as batching and padding are not usually being addressed in this package. Do you think it makes sense to consider padding for this PR, instead of working on it as a more general separate change?